pkg/sbom/scanner/scanner.go

```diff
@@ -269,19 +275,25 @@ func (s *Scanner) performScan(ctx context.Context, request sbom.ScanRequest, col
 
 	scanResult.CreatedAt = createdAt
 	scanResult.Duration = generationDuration
-	return scanResult
+	return &scanResult
 }
 
-func (s *Scanner) handleScanResult(scanResult sbom.ScanResult, request sbom.ScanRequest, collector collectors.Collector) {
+func (s *Scanner) handleScanResult(scanResult *sbom.ScanResult, request sbom.ScanRequest, collector collectors.Collector, errorType string) {
+	if scanResult == nil {
+		telemetry.SBOMFailures.Inc(request.Collector(), request.Type(), "nil_scan_result")
+		log.Errorf("nil scan result for '%s'", request.ID())
+		return
+	}
```

Comment:
This also looks like dead code.
The only place where `handleScanResult` is called is line 248 where `result` is the same as in the previous comment.

```suggestion
```